### ISSUE TITLE
Hello, could you please clarify whether the grid information stored in the geo file contains grid points for the entire globe or only the information for the selected 200-plus points?

### ISSUE THREAD

**Issue by Fccccccc** [Neoyanghc/HiGRN#3]

**Comment by Neoyanghc** [comment]
Thanks for your question. Only the locations of the selected points are needed for calculation and should be stored in the .geo file.

**Comment by Fccccccc** [comment]
![图七](https://github.com/user-attachments/assets/54dee7e9-66dc-4b5e-8a30-e12913e27f56)
![Snipaste_2024-08-14_16-49-02](https://github.com/user-attachments/assets/a3a7ac97-aebd-44a9-84a0-1f4e7b2a8e33)
The first image shows the information of your dataset, and the second image shows the information of my dataset. Could you please clarify what additional features are included in the last item of your three feature dimensions? After debugging, I still encounter errors when running the code with your dataset.

**Comment by Fccccccc** [comment]
![Snipaste_2024-08-14_17-55-48](https://github.com/user-attachments/assets/3d2524b7-3dae-4999-95b9-78e073022d41)
If you still have the config file and the other two files, could you please upload them? Because your dataset has 3 feature dimensions, while my processed dataset has only 1 dimension. I need to check these files to replicate your dataset. Additionally, after debugging, I also encounter issues when running with your dataset.

**Comment by Fccccccc** [comment]
Hello, could I kindly ask you to take another look at the data processing code? The data I processed is causing issues when input into the model, particularly with the way the dataset is divided. Could you please help me find the relevant code? Alternatively, details on the specific format of the processed files and what content they should include would also be very helpful.

**Comment by Neoyanghc** [comment]
I have double-checked the laptop, but unfortunately, it was lost. Apologize for the inconvenience. If you encounter some other problem, I am willing to help.

**Comment by Fccccccc** [comment]
Hello, I have carefully reviewed the contents of your dataset and found several issues. I will list a few points:

The dataset covers five years, which should amount to 1,826 days, but there are only 1,802 days in the data, missing 24 days. See Figure 1.

The data is supposed to be in the form of a sliding window with a window size of 12, moving one day forward each time. See Figure 2. I have split your dataset into x_train, y_train, x_val, y_val, x_test, and y_test, and saved them as text files. In your training set x_train, the first year’s data follows the correct sliding window of 12, moving one day forward each time. However, when sliding to the last day of the 15th year, the subsequent data does not start with the first day of the 16th year, and the data following this point is not continuous in terms of time. See Figures 3 and 4.

In your test set x_test, there is data that also appears in the training set x_train. I have checked, and this data appears only once in the five years. See Figure 5.

Additionally, in your test set x_test, there are values that do not appear in the five years of data. See Figure 6.

Could you please clarify these issues, or is it possible that the dataset was uploaded incorrectly? Thank you.

**Comment by Fccccccc** [comment]
Hello, I have carefully reviewed the contents of your dataset and found several issues. I will list a few points:

The dataset covers five years, which should amount to 1,826 days, but there are only 1,802 days in the data, missing 24 days. See Figure 1.

The data is supposed to be in the form of a sliding window with a window size of 12, moving one day forward each time. See Figure 2. I have split your dataset into x_train, y_train, x_val, y_val, x_test, and y_test, and saved them as text files. In your training set x_train, the first year’s data follows the correct sliding window of 12, moving one day forward each time. However, when sliding to the last day of the 15th year, the subsequent data does not start with the first day of the 16th year, and the data following this point is not continuous in terms of time. See Figures 3 and 4.

In your test set x_test, there is data that also appears in the training set x_train. I have checked, and this data appears only once in the five years. See Figure 5.

Additionally, in your test set x_test, there are values that do not appear in the five years of data. See Figure 6.

Could you please clarify these issues, or is it possible that the dataset was uploaded incorrectly? Thank you.
![问题1](https://github.com/user-attachments/assets/43f134d9-77f2-4486-9e83-2e73a2ab4769)
![图11](https://github.com/user-attachments/assets/57267c74-98dd-413b-88d8-305952a4fc2f)
![15年最后一天的数据](https://github.com/user-attachments/assets/fe6d04c7-ffb5-4ea1-a52c-ea1e6d5520e5)
![16年第一天的数据](https://github.com/user-attachments/assets/fb590fc8-c0ce-4afe-bebe-5070ed8f25c3)
![数据集问题](https://github.com/user-attachments/assets/fd865932-6934-44ee-865e-690489d88922)
![数据集问题3](https://github.com/user-attachments/assets/189930d7-fe40-4fbb-a762-0daf884c7821)

**Comment by Fccccccc** [comment]
Hi. There's another issue, which is the adjacency matrix. Since the adjacency matrix initialized in the code is all zeros, and remains all zeros after processing, it doesn't serve any purpose. So, do we need to input the adjacency matrix separately?

**Comment by Fccccccc** [comment]
There's another issue, which is the adjacency matrix. Since the adjacency matrix initialized in the code is all zeros, and remains all zeros after processing, it doesn't serve any purpose. So, do we need to input the adjacency matrix separately?

**Comment by Neoyanghc** [comment]
It is initialized randomly by the “def calculate_scaled_laplacian(adj)” and updated the parameters by the loss. Our method does not need the input matrix but lets it be self-learnable to capture the dynamic correlations to solve the problem.

**Comment by Fccccccc** [comment]
Hello, sorry to bother you so late. I have two questions.

1、During my parameter tuning process, everything works fine when the graph_layer is set to 1. However, when I increase the graph_layer to 2, the number of layers (num_layers) can only be set to a very small value, like 2 or 3. If I increase it further, I get an out-of-memory error on the GPU. The error message is as follows. I'm using a 4090 GPU. Did you encounter this issue when you used this set of parameters before?
File "/data/run01/scw6d2w/HiGRN-main/lib/model/HiGRU/HiGRU.py", line 258, in forward

    x_gconv = torch.einsum('bnki,nkioc->bnoc', x_g, self.cluster_weights_pool)  + self.bias_pool    # b, N, dim_out, cluster_dim

torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 22.00 MiB. GPU 0 has a total capacity of 23.65 GiB of which 19.25 MiB is free. Including non-PyTorch memory, this process has 23.62 GiB memory in use. Of the allocated memory 23.13 GiB is allocated by PyTorch, and 44.14 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
![Snipaste_2024-08-26_23-31-30](https://github.com/user-attachments/assets/e1d25286-ac6e-4867-a21a-191c2686a79b)

2、How did you create the image in your paper? I'd like to differentiate the nodes in two regions in a similar way in my image.
![Snipaste_2024-08-26_23-32-08](https://github.com/user-attachments/assets/e71d669e-9463-4b39-92a2-a86866bdfb81)

**Comment by Neoyanghc** [comment]
Out-of-memory problems are really common. When you have more data to train, you can try to use multiple GPUs. For the Figure, I utilized T-she to draw.
